### PR TITLE
feat: Landscape directory mapbox migration

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Landscape directory mapbox migration

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added
- [ ] English strings reviewed and copyedited
- [ ] Strings are localized
- [ ] Images are compressed (TinyPNG or svgo)
- [ ] Sample environment file updated (when environment variables changed)
- [ ] Verified on mobile
- [ ] Verified on desktop
- [ ] Verified accessibility ([Accessibility - Web applications](https://docs.google.com/document/d/1VwPDyLqw7r_iISMgDZr1FdAmAqqOIIVIxQqsYA-HaHE/edit?usp=sharing))

### Related Issues
Fixes https://github.com/techmatters/terraso-product/issues/120

### Verification steps

image

### CODE REVIEW

File: src/storyMap/components/StoryMapForm/MapLocationDialog.js [hunk 1]
```diff
@@ -15,34 +15,55 @@
...
+    map.on('load', updatePosition);
+    map.on('move', updatePosition);
+
+    return () => {
+      map.off('load', updatePosition);
+      map.off('move', updatePosition);
+    };
+  }, [map, onPositionChange]);
+
+  return null;
```
Comment (paulschreiber) [inline src/storyMap/components/StoryMapForm/MapLocationDialog.js:59]: question: is this required? if so, should it match the return on line 40?

File: src/storyMap/components/StoryMap.js [hunk 2]
```diff
@@ -218,16 +284,23 @@ const StoryMap = props => {
 
   const getLayerPaintType = useCallback(
     layer => {
-      const layerType = map.getLayer(layer).type;
+      const layerType = map.getLayer(layer)?.type;
+      if (!layerType) {
+        console.warn(`Layer ${layer} not found`);
```
Comment (paulschreiber) [inline src/storyMap/components/StoryMap.js:289]: question: should this be removed?

File: src/storyMap/components/StoryMap.js [hunk 3]
```diff
@@ -218,16 +284,23 @@ const StoryMap = props => {
...
+        return null;
```
Comment (paulschreiber) [inline src/storyMap/components/StoryMap.js:290]: question: should this be `return` or `return null`?

File: src/storyMap/components/StoryMap.js [hunk 4]
```diff
@@ -190,24 +185,95 @@ const getTransition = (config, id) => {
...
+      });
+      // Add a black outline around the polygon.
+      insetMap.addLayer({
+        id: 'outlineLayer',
+        type: 'line',
+        source: 'boundsSource',
+        layout: {},
+        paint: {
+          'line-color': theme.palette.black,
+          'line-width': 1,
```
Comment (paulschreiber) [inline src/storyMap/components/StoryMap.js:220]: question: should this be in a constant or configuration variable?

File: src/landscape/components/LandscapeListMap.js [hunk 5]
```diff
@@ -83,55 +73,169 @@ const LandscapesClusters = props => {
...
+
+    map.on('mouseenter', 'clusters', () => {
+      map.getCanvas().style.cursor = 'pointer';
+    });
+    map.on('mouseleave', 'clusters', () => {
+      map.getCanvas().style.cursor = '';
+    });
+    map.on('mouseenter', 'unclustered-point', () => {
+      map.getCanvas().style.cursor = 'pointer';
+    });
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeListMap.js:189]: question: can this be combined with the same code on line 181?

File: src/landscape/components/LandscapeListMap.js [hunk 6]
```diff
@@ -83,55 +73,169 @@ const LandscapesClusters = props => {
...
+
+    map.addLayer({
+      id: 'cluster-count',
+      type: 'symbol',
+      source: 'landscapes',
+      filter: ['has', 'point_count'],
+      layout: {
+        'text-field': ['get', 'point_count_abbreviated'],
+        'text-font': ['DIN Offc Pro Medium', 'Arial Unicode MS Bold'],
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeListMap.js:121]: question: where does this font stack come from?

File: src/landscape/components/LandscapeList.test.js [hunk 7]
```diff
@@ -101,19 +149,32 @@ const baseListTest = async () => {
...
+      features: [geojson.features[0]],
+      lngLat: {
+        lng: 0,
+        lat: 0,
+      },
+    })
+  );
+  const domElement = Popup.setDOMContent.mock.calls[0][0];
+  expect(domElement.querySelector('a').href).toEqual(
+    'http://localhost/landscapes/landscape-0'
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeList.test.js:175]: question: should this be localhost or 127.0.0.1?

File: src/gis/components/MapboxGeocoder.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      return;
+    }
+    const geocoder = new MapboxGlGeocoder({
+      accessToken: MAPBOX_ACCESS_TOKEN,
+      placeholder: t('storyMap.form_location_dialog_geocoder_placeholder'),
+      mapboxgl,
+    });
+    map.addControl(geocoder);
+  }, [map, t]);
+  return null;
```
Comment (paulschreiber) [inline src/gis/components/MapboxGeocoder.js:38]: question: same a below: should this be return null or return?

**Review by paulschreiber — COMMENTED** [review]
This looks good. There's a lot of inline style configuration. Does it get reused? If so, where is the best place to store it?

File: src/storyMap/components/StoryMapForm/MapLocationDialog.js (hunk 1 above)
Comment (josebui) [reply src/storyMap/components/StoryMapForm/MapLocationDialog.js:59]: Yes, a React component can return `undefined`

File: src/storyMap/components/StoryMap.js (hunk 2 above)
Comment (josebui) [reply src/storyMap/components/StoryMap.js:289]: I think it should be there, I think I can improve it using the `logger`, this means that a laye added in the configuration doesn't exist in the mapbox style

File: src/landscape/components/LandscapeListMap.js (hunk 6 above)
Comment (josebui) [reply src/landscape/components/LandscapeListMap.js:121]: I have to check I think ideally we should use the same fonts as the rest of the site, so I will see if I can remove them

File: src/landscape/components/LandscapeList.test.js (hunk 7 above)
Comment (josebui) [reply src/landscape/components/LandscapeList.test.js:175]: Not, sure, the test is passing so I think that is how the url works for the tests, but I will double check this

File: src/storyMap/components/StoryMap.js (hunk 4 above)
Comment (josebui) [reply src/storyMap/components/StoryMap.js:220]: I don't think this belongs in a constant or configurable value, this is similar to the inline styling we have for any component, we don't have constant for that, and also this is not being reused

**Comment by josebui** [comment]
> This looks good. There's a lot of inline style configuration. Does it get reused? If so, where is the best place to store it?

@paulschreiber Can you add more context to this? where do you see the duplicated styles?

**Comment by paulschreiber** [comment]
@josebui I don't see style information duplicated anywhere. I'm just wondering if it make sense to put it somewhere other then in the midst of large code block.

**Comment by josebui** [comment]
> @josebui I don't see style information duplicated anywhere. I'm just wondering if it make sense to put it somewhere other then in the midst of large code block.

@paulschreiber There are some changes of code moving as part of this PR, so a lot of the changes you see here are not related to the scope of this task, I don't see the large blocks of inline styling, if you can point me to them maybe I can see how we can improve that, still it seems out of scope fot this task

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**